File: Program/LawCreator/test/test/lawcreator/xml/nodes/NodeTest.java
package test.lawcreator.xml.nodes;

import static org.junit.Assert.*;

import org.junit.Before;
import org.junit.Test;

import pl.wroc.pwr.student.lawcreator.xml.nodes.Attribute;
import pl.wroc.pwr.student.lawcreator.xml.nodes.Node;

public class NodeTest {
	Node node;

	@Before
	public void setUp() throws Exception {
		node = new Node();
	}

	@Test
	public void gettingAndSettingFieldsTest() {
		String name = "name";
		String value = "value";
		
		node.setName(name);
		node.setValue(value);
		
		assertEquals(name, node.getName());
		assertEquals(value, node.getValue());
	}
	
	@Test
	public void addingAttributesTest() {
		Attribute att1 = new Attribute();
		att1.setName("att1");
		att1.setValue("v1");
		Attribute att2 = new Attribute();
		att2.setName("att2");
		att2.setValue("v2");
		
		node.addAttribute(att1);
		node.addAttribute(att2);
		
		assertEquals(2, node.getAttributes().size());
		assertEquals(att1, node.getAttributeByName("att1"));
		assertEquals(att2, node.getAttributeByName("att2"));
	}
	
	@Test
	public void addingNodesTest() {
		Node child1 = new Node();
		child1.setName("ch1");
		child1.setValue("v1");
		Node child2 = new Node();
		child2.setName("ch2");
		child2.setValue("v2");
		
		node.addNote(child1);
		node.addNote(child2);
		
		assertEquals(2, node.getNodes().size());
		assertEquals(child1, node.getChildByName("ch1"));
		assertEquals(child2, node.getChildByName("ch2"));
	}
	
	@Test
	public void gettingXMLTest() {
		Attribute att1 = new Attribute();
		att1.setName("att1");
		att1.setValue("v1");
		Attribute att2 = new Attribute();
		att2.setName("att2");
		att2.setValue("v2");
		Attribute att3 = new Attribute();
		att3.setName("att3");
		att3.setValue("v3");
		Node child1 = new Node();
		child1.setName("ch1");
		child1.setValue("v1");
		child1.addAttribute(att1);
		Node child2 = new Node();
		child2.setName("ch2");
		child2.setValue("v2");
		child2.addAttribute(att2);
		node.addAttribute(att3);
		node.addNote(child1);
		node.addNote(child2);
		
		String xml = node.toXML();

		assertTrue(xml.contains("<ch1 att1=\"v1\">\nv1\n</ch1>\n<ch2" +
					"att2=\"v2\">\nv2\n</ch2>"));
	}

}

File: Program/LawCreator/bin/test/lawcreator/xml/nodes/NodeTest.js
Clazz.declarePackage ("test.lawcreator.xml.nodes");
Clazz.load (null, "test.lawcreator.xml.nodes.NodeTest", ["pl.wroc.pwr.student.lawcreator.xml.nodes.Attribute", "$.Node"], function () {
c$ = Clazz.decorateAsClass (function () {
this.node = null;
Clazz.instantialize (this, arguments);
}, test.lawcreator.xml.nodes, "NodeTest");
Clazz.defineMethod (c$, "setUp", 
function () {
this.node =  new pl.wroc.pwr.student.lawcreator.xml.nodes.Node ();
});
Clazz.defineMethod (c$, "gettingAndSettingFieldsTest", 
function () {
var name = "name";
var value = "value";
this.node.setName (name);
this.node.setValue (value);
org.junit.Assert.assertEquals (name, this.node.getName ());
org.junit.Assert.assertEquals (value, this.node.getValue ());
});
Clazz.defineMethod (c$, "addingAttributesTest", 
function () {
var att1 =  new pl.wroc.pwr.student.lawcreator.xml.nodes.Attribute ();
att1.setName ("att1");
att1.setValue ("v1");
var att2 =  new pl.wroc.pwr.student.lawcreator.xml.nodes.Attribute ();
att2.setName ("att2");
att2.setValue ("v2");
this.node.addAttribute (att1);
this.node.addAttribute (att2);
org.junit.Assert.assertEquals (2, this.node.getAttributes ().size ());
org.junit.Assert.assertEquals (att1, this.node.getAttributeByName ("att1"));
org.junit.Assert.assertEquals (att2, this.node.getAttributeByName ("att2"));
});
Clazz.defineMethod (c$, "addingNodesTest", 
function () {
var child1 =  new pl.wroc.pwr.student.lawcreator.xml.nodes.Node ();
child1.setName ("ch1");
child1.setValue ("v1");
var child2 =  new pl.wroc.pwr.student.lawcreator.xml.nodes.Node ();
child2.setName ("ch2");
child2.setValue ("v2");
this.node.addNote (child1);
this.node.addNote (child2);
org.junit.Assert.assertEquals (2, this.node.getNodes ().size ());
org.junit.Assert.assertEquals (child1, this.node.getChildByName ("ch1"));
org.junit.Assert.assertEquals (child2, this.node.getChildByName ("ch2"));
});
Clazz.defineMethod (c$, "gettingXMLTest", 
function () {
var att1 =  new pl.wroc.pwr.student.lawcreator.xml.nodes.Attribute ();
att1.setName ("att1");
att1.setValue ("v1");
var att2 =  new pl.wroc.pwr.student.lawcreator.xml.nodes.Attribute ();
att2.setName ("att2");
att2.setValue ("v2");
var att3 =  new pl.wroc.pwr.student.lawcreator.xml.nodes.Attribute ();
att3.setName ("att3");
att3.setValue ("v3");
var child1 =  new pl.wroc.pwr.student.lawcreator.xml.nodes.Node ();
child1.setName ("ch1");
child1.setValue ("v1");
child1.addAttribute (att1);
var child2 =  new pl.wroc.pwr.student.lawcreator.xml.nodes.Node ();
child2.setName ("ch2");
child2.setValue ("v2");
child2.addAttribute (att2);
this.node.addAttribute (att3);
this.node.addNote (child1);
this.node.addNote (child2);
var xml = this.node.toXML ();
org.junit.Assert.assertTrue (xml.contains ("<ch1 att1=\"v1\">\nv1\n</ch1>\n<ch2att2=\"v2\">\nv2\n</ch2>"));
});
});
